src/tree/AzureDBAPIStep.ts

```diff
@@ -0,0 +1,70 @@
+import { PostgresServerFirewallStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerFirewallStep';
+import { PostgresServerNameStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerNameStep';
+import { PostgresServerSetCredentialsStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerSetCredentialsStep';
+import { PostgresServerSetFirewallStep } from '../postgres/commands/PostgresAccountWizard/PostgresServerSetFirewallStep';
+import { localize } from '../utils/localize';
+import { CosmosDBAccountCreateStep } from './CosmosDBAccountWizard/CosmosDBAccountCreateStep';
+import { CosmosDBAccountNameStep } from './CosmosDBAccountWizard/CosmosDBAccountNameStep';
+import { ICosmosDBWizardContext } from './CosmosDBAccountWizard/ICosmosDBWizardContext';
+import { IAzureDBWizardContext } from './IAzureDBWizardContext';
+
+export class AzureDBAPIStep extends AzureWizardPromptStep<IPostgresWizardContext | ICosmosDBWizardContext> {
+    public async prompt(wizardContext: IAzureDBWizardContext): Promise<void> {
+        const picks: IAzureQuickPickItem<Experience>[] = getExperienceQuickPicks();
+
+        const result: IAzureQuickPickItem<Experience> = await ext.ui.showQuickPick(picks, {
+            placeHolder: localize('selectDBServerMsg', 'Select an Azure Database Server.')
+        });
+
+        wizardContext.defaultExperience = result.data;
+    }
+
+    public async getSubWizard(wizardContext: IAzureDBWizardContext): Promise<IWizardOptions<IPostgresWizardContext | ICosmosDBWizardContext>> {
+        let promptSteps: AzureWizardPromptStep<IPostgresWizardContext | ICosmosDBWizardContext>[];
+        let executeSteps: AzureWizardExecuteStep<IPostgresWizardContext | ICosmosDBWizardContext>[];
+        if (wizardContext.defaultExperience?.api === API.Postgres) {
+            promptSteps = [
+                new PostgresServerNameStep(),
+                new PostgresServerCredUserStep(),
+                new PostgresServerCredPWStep(),
+                new PostgresServerConfirmPWStep(),
+                new PostgresServerFirewallStep()
+            ];
+            executeSteps = [
+                new PostgresServerCreateStep(),
+                new PostgresServerSetCredentialsStep(),
+                new PostgresServerSetFirewallStep()
+            ];
+            // tslint:disable-next-line: no-unnecessary-local-variable
+            const wizardOptions: IWizardOptions<IPostgresWizardContext> = { promptSteps, executeSteps };
+            return wizardOptions;
+        } else {
+            promptSteps = [
+                new CosmosDBAccountNameStep()
+            ];
+            executeSteps = [
+                new CosmosDBAccountCreateStep()
+            ];
+            // tslint:disable-next-line: no-unnecessary-local-variable
+            const wizardOptions: IWizardOptions<ICosmosDBWizardContext> = { promptSteps, executeSteps };
+            return wizardOptions;
```

Comment:
😝 I feel like you're a little too quick to disable tslint rules sometimes. It's saying the variable `wizardOptions` is unnecessary and you could just do this instead:
```
return { promptSteps, executeSteps };
```

I feel like that looks a lot cleaner

Reply:
I was having some problems with type casting in beginning, wizardOptions was the fix and 
I added that to this commit as well...don't really need it now 😅